v3/sockjs/options.go

```diff
@@ -42,7 +42,7 @@ type Options struct {
 	RawWebsocket bool
 	// Provide a custom Upgrader for Websocket connections to enable features like compression.
 	// See https://godoc.org/github.com/gorilla/websocket#Upgrader for more details.
-	WebsocketUpgrader *websocket.Upgrader
+	WebsocketUpgrader upgrader
```

Comment:
```suggestion
	WebsocketUpgrader func(w http.ResponseWriter, r *http.Request, responseHeader http.Header) (*websocket.Conn, error)
```

would this also work for you? In that case we would not need to create an interface

Reply:
but your change is backwards compatible, so probably better choice in this case

Reply:
Yes function maybe more clean solution, but, backward compatible.